spec/support/test_db.rb

```diff
@@ -0,0 +1,146 @@
+class TestDb
+
+  DATABASE_USERS = %w{hr arunit}
+
+  def self.build
+    db = self.new
+    db.cleanup_database_users
+    db.create_user_tablespace
+    db.setup_database_users
+    db.connection.logoff
+  end
+
+  def self.database_version
+     db = self.new
+     db.database_version
+  end
+
+  def connection
+    unless defined?(@connection)
+      begin
+        Timeout::timeout(5) {
+          if defined?(JRUBY_VERSION)
+            @connection = java.sql.DriverManager.get_connection(
+              'jdbc:oracle:thin:@127.0.0.1:1521/XE',
+              'system',
+              'oracle'
+            );
+          else
+            @connection = OCI8.new(
+              'system',
+              'oracle',
+              '(DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=127.0.0.1)(PORT=1521))(CONNECT_DATA=(SERVICE_NAME=XE)))'
+            )
+          end
+        }
+      rescue Timeout::Error
+        raise "Cannot establish connection with Oracle database as SYSTEM user. Seams you need to start local Oracle database"
+      end
+    end
+    @connection
+  end
+
+  def create_user_tablespace
+    return unless connection
+    execute_statement(<<-STATEMENT
+      DECLARE
+        v_exists number;
+      BEGIN
+        SELECT count(1)
+          INTO v_exists
+          FROM dba_tablespaces
+          WHERE tablespace_name = 'TBS_USERS';
+
+          IF v_exists = 0 THEN
+            EXECUTE IMMEDIATE 'ALTER SYSTEM SET DB_CREATE_FILE_DEST = ''/u01/app/oracle/oradata/XE''';
+            EXECUTE IMMEDIATE 'CREATE TABLESPACE TBS_USERS DATAFILE ''tbs_users.dat'' SIZE 10M REUSE AUTOEXTEND ON NEXT 10M MAXSIZE 200M';
+          END IF;
+      END;
+      STATEMENT
+      )
+  end
+
+  def cleanup_database_users
+    return unless connection
+    DATABASE_USERS.each do |db|
```

Comment:
I think it would be cleaner to move the Upper Case conversion here (instead of calling UPPER inside the SQL).